server/src/main/scala/io/delta/kernelsharedtable/DeltaSharedTableKernel.scala

```diff
@@ -308,6 +364,145 @@ class DeltaSharedTableKernel(
     throw new DeltaSharingUnsupportedOperationException("not implemented yet")
   }
 
+  private def processBatchByColumnVector(
+      scanFileBatch: FilteredColumnarBatch,
+      version: Long,
+      queryType: QueryTypes.QueryType,
+      respondedFormat: String,
+      table: Table,
+      engine: Engine,
+      isVersionQuery: Boolean,
+      snapshot: SharedTableSnapshot): Seq[Object] = {
+
+    val batchData = scanFileBatch.getData
+    val batchSize = batchData.getSize
+    val addFileVectors = getAddFileColumnVectors(batchData)
+    val selectionVector = scanFileBatch.getSelectionVector
+    var addFileObjects = Seq[Object]()
+    for (rowId <- 0 until batchSize) {
+      val isSelected = !selectionVector.isPresent ||
+        (!selectionVector.get.isNullAt(rowId) && selectionVector.get.getBoolean(rowId))
+      if (isSelected) {
+        val partitionValues = getDeltaPartitionValues(addFileVectors.partitionValues, rowId)
```

Comment:
Why are you performing some operations outside `getResponseAddFile` and some inside it? 
Considering putting them together?

Reply:
not too sure if i understand what you mean here. I moved some duplicate logic out of getResponseAddFile in latest commit

Reply:
As we are passing `addFileVectorPath` into `getResponseAddFile`, I think we can move L386-L389 all into `getResponseAddFile`

Reply:
fixed